### PR TITLE
feat: unify pipeline trigger chart endpoint with credit chart

### PR DESCRIPTION
**Pull request by jvallesm**
Because

- Pipeline and credit endpoints have different params and schemas.
- Some of the information served by the existing endpoints will be handled by the logging feature.

This commit

- Applies contracts defined in https://github.com/instill-ai/protobufs/pull/387

### CODE REVIEW

**Comment by jvallesm** [comment]
# QA

```sh
$ curl 'localhost:8080/v1beta/metrics/credit/charts?namespaceId=jvallesm&start=2024-07-22T22:00:00.000Z&stop=2024-07-23T22:00:00.000Z&aggregationWindow=1h' \
  -H 'Authorization: Bearer instill_sk_1Myb31r6YsjVt8In2zf8vRvq6wUpfvRq' | jq


  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   754  100   754    0     0  18113      0 --:--:-- --:--:-- --:--:-- 18390
{
  "creditConsumptionChartRecords": [
    {
      "namespaceId": "",
      "timeBuckets": [
        "2024-07-22T22:00:00Z",
        "2024-07-22T23:00:00Z",
        "2024-07-23T00:00:00Z",
        "2024-07-23T01:00:00Z",
        "2024-07-23T02:00:00Z",
        "2024-07-23T03:00:00Z",
        "2024-07-23T04:00:00Z",
        "2024-07-23T05:00:00Z",
        "2024-07-23T06:00:00Z",
        "2024-07-23T07:00:00Z",
        "2024-07-23T08:00:00Z",
        "2024-07-23T09:00:00Z",
        "2024-07-23T10:00:00Z",
        "2024-07-23T11:00:00Z",
        "2024-07-23T12:00:00Z",
        "2024-07-23T13:00:00Z",
        "2024-07-23T14:00:00Z",
        "2024-07-23T15:00:00Z",
        "2024-07-23T16:00:00Z",
        "2024-07-23T17:00:00Z",
        "2024-07-23T18:00:00Z",
        "2024-07-23T19:00:00Z",
        "2024-07-23T20:00:00Z",
        "2024-07-23T21:00:00Z"
      ],
      "amount": [
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        64.81,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0,
        0
      ],
      "source": "pipeline"
    }
  ]
}
```

**Comment by jvallesm** [comment]
# QA

```sh
$ curl 'localhost:8080/v1beta/metrics/vdp/pipeline/trigger-count?namespaceId=jvallesm&start=2024-07-22T22:00:00.000Z&stop=2024-07-25T22:00:00.000Z' \
  -H 'Authorization: Bearer instill_sk_1Myb31r6YsjVt8In2zf8vRvq6wUpfvRq' | jq

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   122  100   122    0     0   7058      0 --:--:-- --:--:-- --:--:--  7176
{
  "pipelineTriggerCounts": [
    {
      "triggerCount": 8,
      "status": "STATUS_COMPLETED"
    },
    {
      "triggerCount": 2,
      "status": "STATUS_ERRORED"
    }
  ]
}
```

File: pkg/repository/repository.go [hunk 1]
```diff
@@ -61,10 +61,11 @@ type Repository interface {
...
 
+	GetOwner(ctx context.Context, id string, includeAvatar bool) (*datamodel.Owner, error)
+	GetOwnerByUID(ctx context.Context, uid uuid.UUID) (*datamodel.Owner, error)
+
 	listOwners(ctx context.Context, ownerType string, pageSize int, pageToken string, filter filtering.Filter) ([]*datamodel.Owner, int64, string, error)
 	createOwner(ctx context.Context, ownerType string, user *datamodel.Owner) error
-	getOwner(ctx context.Context, ownerType string, id string, includeAvatar bool) (*datamodel.Owner, error)
-	getOwnerByUID(ctx context.Context, ownerType string, uid uuid.UUID) (*datamodel.Owner, error)
 	updateOwner(ctx context.Context, ownerType string, id string, user *datamodel.Owner) error
 	deleteOwner(ctx context.Context, ownerType string, id string) error
```
Comment (jvallesm) [inline pkg/repository/repository.go:70]: 💭 @donch1989 if owner ID is unique perhaps we don't need the type checks in these methods and we can delegate on the service layer to check the authenticated user permissions to perform these actions.

File: pkg/repository/repository.go [hunk 2]
```diff
@@ -250,12 +285,11 @@ func (r *repository) createOwner(ctx context.Context, ownerType string, owner *d
 	return nil
 }
 
-func (r *repository) getOwner(ctx context.Context, ownerType string, id string, includeAvatar bool) (*datamodel.Owner, error) {
-
+func (r *repository) GetOwner(ctx context.Context, id string, includeAvatar bool) (*datamodel.Owner, error) {
```
Comment (jvallesm) [inline pkg/repository/repository.go:288]: @donch1989 I mentioned this change in your PR yesterday, what do you think?

File: pkg/repository/repository.go [hunk 3]
```diff
@@ -128,12 +160,25 @@ func (r *repository) ListOrganizations(ctx context.Context, pageSize int, pageTo
 func (r *repository) CreateOrganization(ctx context.Context, org *datamodel.Owner) error {
 	return r.createOwner(ctx, "organization", org)
 }
+
 func (r *repository) GetOrganization(ctx context.Context, id string, includeAvatar bool) (*datamodel.Owner, error) {
-	return r.getOwner(ctx, "organization", id, includeAvatar)
+	owner, err := r.GetOwner(ctx, id, includeAvatar)
```
Comment (donch1989) [inline pkg/repository/repository.go:165]: Following the question [here](https://github.com/instill-ai/mgmt-backend/pull/228/files#r1691606478):

What happens if we use a user ID with the `GetOrganization` function?

File: pkg/repository/repository.go (hunk 2 above)
Comment (donch1989) [reply pkg/repository/repository.go:288]: Yeah, we can expose it.

**Review by donch1989 — APPROVED** [review]
LGTM

**Pull request merged**